services/manifest-repo-export-service/pkg/repository/transformer.go

```diff
@@ -2139,10 +2125,76 @@ func (c *DeleteAAEnvironmentConfig) GetEslVersion() db.TransformerID {
 }
 
 func (c *DeleteAAEnvironmentConfig) Transform(
-	_ context.Context,
-	_ *State,
-	_ TransformerContext,
+	ctx context.Context,
+	state *State,
+	tCtx TransformerContext,
 	_ *sql.Tx,
 ) (string, error) {
-	return GetNoOpMessage(c)
+	fs := state.Filesystem
+
+	envDir := fs.Join("environments", string(c.Environment))
+	if err := fs.MkdirAll(envDir, 0777); err != nil {
+		return "", err
+	}
+
+	configFile := fs.Join(envDir, "config.json")
+
+	data, err := util.ReadFile(fs, configFile)
+	if err != nil {
+		return "", fmt.Errorf("error opening file: %w", err)
+	}
+	var envConfig config.EnvironmentConfig
+
+	err = json.Unmarshal(data, &envConfig)
+	if err != nil {
+		return "", err
+	}
+
+	argoCdAppFile := fs.Join("argocd", string(argocd.V1Alpha1), fmt.Sprintf("%s.yaml", *envConfig.ArgoCdConfigs.CommonEnvPrefix+"-"+string(c.Environment)+"-"+string(c.ConcreteEnvironmentName)))
+
+	err = fs.Remove(argoCdAppFile)
+	if errors.Is(err, os.ErrNotExist) {
+		logger.FromContext(ctx).Sugar().Warnf("AA environment argocd app file %q does not exist.", argoCdAppFile)
+	} else if err != nil {
+		return "", fmt.Errorf("error deleting AA environment's argocd app file %q: %w", argoCdAppFile, err)
+	}
+
+	if tCtx.ShouldMinimizeGitData() {
+		return fmt.Sprintf("removed configuration for AA environment %q - %q", c.Environment, c.ConcreteEnvironmentName), nil
+	}
+
+	for idx, currentConfig := range envConfig.ArgoCdConfigs.ArgoCdConfigurations {
+		if types.EnvName(currentConfig.ConcreteEnvName) == c.ConcreteEnvironmentName {
+			envConfig.ArgoCdConfigs.ArgoCdConfigurations = append(envConfig.ArgoCdConfigs.ArgoCdConfigurations[:idx], envConfig.ArgoCdConfigs.ArgoCdConfigurations[idx+1:]...)
+			break
+		}
+	}
+
+	err = writeEnvironmentConfigurationToManifestRepo(fs, configFile, envConfig)
+	if err != nil {
+		return "", fmt.Errorf("error writing environment configuration to manifest repository: %w", err)
+	}
+	return fmt.Sprintf("removed configuration for AA environment %q - %q", c.Environment, c.ConcreteEnvironmentName), nil
+}
+
+func writeEnvironmentConfigurationToManifestRepo(fs billy.Filesystem, configFile string, envConfig config.EnvironmentConfig) error {
+	err := fs.Remove(configFile) // New config was just getting appended to the old configuration . Removing the file and recreating it resulted in the correct output in the file.
```

Comment:
The comment could be clearer
```suggestion
	err := fs.Remove(configFile) // If we do not remove the file, we get an "append" not "overwrite". Removing the file and recreating it resulted in the correct output in the file.
```